### PR TITLE
Pre-trained depth encoder

### PR DESCRIPTION
**Pull request by gorjanradevski**

### CODE REVIEW

**Comment by gorjanradevski** [comment]
We need to have a look at the pre-computed features, because the output from the pretrained depth encoder (https://github.com/jacobkrantz/VLN-CE/blob/master/vlnce_baselines/models/encoders/resnet_encoders.py#L26) seems to return a tensor of size [batch_size, 128] while the pre-computed features are of shape [batch_size, 4, 4, 128].

**Comment by gorjanradevski** [comment]
I think that all is fine now, maybe someone can skim over the changes before merging

File: src/ai/base_net.py [hunk 1]
```diff
@@ -52,12 +49,11 @@ class ArchitectureConfig(Config):
     depth_features_size: int = 2048
     hidden_size: int = 128
     instruction_encoder_layers: int = 1
-    bidirectional_instruction_encoder: bool = False 
+    bidirectional_instruction_encoder: bool = False
     final_state_only_instruction_encoder: bool = True
-    depth_encoder_checkpoint_path: str = ""
+    depth_encoder_checkpoint_path: str = "models/ddppo-models/gibson-2plus-resnet50.pth"
```
Comment (kkelchte) [inline src/ai/base_net.py:54]: Could you make sure that this models/ddppo-models/... exists on gluster/visics/vln/data with correct permissions?

File: src/data/test/test_data_saver.py [hunk 2]
```diff
@@ -1,85 +1,128 @@
...
+    action: int = None,
+) -> Tuple[str, str]:
+    config_dict = {"output_path": output_dir, "max_size": 9, "tag": "test"}
+    config = DataSaverConfig().create(config_dict=config_dict)
+    data_saver = DataSaver(config=config)
+    for _ in range(3):
+        for _ in range(10):
+            experience = Experience(
+                observation={
+                    "rgb_raw": np.zeros((3, 224, 224)),
```
Comment (kkelchte) [inline src/data/test/test_data_saver.py:27]: The dataset generated should be (224, 224, 3) as this is what the simulator provides. The transform should permute accordingly (or even change size as I guess the simulator/dataset is not necessarily adjusted to the observation sizes of rgb_raw.

File: src/data/test/test_data_saver.py [hunk 3]
```diff
@@ -1,85 +1,128 @@
...
+                        "instruction_text": "turn around in circles",
+                        "instruction_tokens": [5, 15, 1],
+                    },
+                }
+            }
+            if with_instruction
+            else None,
         )
         datasaver.save(experience)
     datasaver.dump()
```
Comment (kkelchte) [inline src/data/test/test_data_saver.py:77]: We won't need run_episode anymore as this is fused into generate_dataset

**Review by kkelchte — CHANGES_REQUESTED** [review]
Left some comments which should be shortly dealt with first

File: src/ai/base_net.py (hunk 1 above)
Comment (gorjanradevski) [reply src/ai/base_net.py:54]: Done.

File: src/data/test/test_data_saver.py (hunk 2 above)
Comment (gorjanradevski) [reply src/data/test/test_data_saver.py:27]: yes, my bad, fixed now.

File: src/data/test/test_data_saver.py (hunk 3 above)
Comment (gorjanradevski) [reply src/data/test/test_data_saver.py:77]: do you want me to delete it?

File: src/ai/base_net.py (hunk 1 above)
Comment (kkelchte) [reply src/ai/base_net.py:54]: have no access...
[kkelchte@opal data]$ ls -lt models
totaal 4
drwx------ 2 gradevsk visics 4096 24 mrt 15:57 ddppo-models
[kkelchte@opal data]$

File: src/data/test/test_data_saver.py (hunk 3 above)
Comment (kkelchte) [reply src/data/test/test_data_saver.py]: yeah sure

**Review by kkelchte — APPROVED** [review]
given you adjust permission to model, this is good to merge!

**Pull request merged**